src/main/java/command/ClearCommand.java

```diff
@@ -2,27 +2,34 @@
 
 import cheatsheet.CheatSheetList;
 import storage.DataFileDestroyer;
+import storage.DataFileReader;
 import ui.Printer;
 import ui.Ui;
 
 public class ClearCommand extends Command {
     protected DataFileDestroyer fileDestroyer;
+    protected DataFileReader fileReader;
 
     public static final String invoker = "/clear";
 
-    public ClearCommand(Printer printer, CheatSheetList cheatSheetList, DataFileDestroyer fileDestroyer) {
+    public ClearCommand(Printer printer, CheatSheetList cheatSheetList,
+                        DataFileDestroyer fileDestroyer, DataFileReader fileReader) {
         super(printer);
         this.cheatSheetList = cheatSheetList;
         this.fileDestroyer = fileDestroyer;
+        this.fileReader = fileReader;
     }
 
     @Override
     public void execute() {
         printer.printClearConfirmation();
         if (isClearConfirmed()) {
+            final int deletedCheatSheets = cheatSheetList.getSize();
             fileDestroyer.executeFunction();
-            printer.printClearCheatSheetMessage(cheatSheetList.getSize());
             cheatSheetList.clear();
+            fileReader.extractPreloadedCheatSheets();
+            fileReader.executeFunction();
+            printer.printClearCheatSheetMessage(deletedCheatSheets);
```

Comment:
I think this one should be 
`printer.printClearCheatSheetMessage(deletedCheatSheets - cheatSheetList.getSize())` or something equivalent since we don't really "delete" the preloaded ones.

Reply:
Noted! Will do the necessary amendments